source/common/unified_malloc_framework/src/ipc.c

```diff
@@ -0,0 +1,100 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+ * See LICENSE.TXT
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include "umf/ipc.h"
+
+#include "memory_pool_internal.h"
+#include "memory_tracker.h"
+
+#include <assert.h>
+#include <stdlib.h>
+
+struct umf_ipc_data_t {
+    uint32_t size;
```

Comment:
why not uint64_t?

Reply:
this variable holds the size of the `umf_ipc_data_t` data structure. It is hardly possible that it is big. So I used `uint32_t` to decrease the size of the IPC handle. Now I am thinking that even `uint16_t` should be enough.

Reply:
The `umf_ipc_data_t` will still be 16B, no matter if you use `uint32_t`, `uint16_t` or `uint64_t` due to alignment requirements.

Reply:
good point. So there are two possible ways, use special pragmas to forbid the compiler from adding implicit padding or like you said use uint64_t. I think for now we can just simplify and use uint64_t and in the future, if we have a requirement to decrease the size of the handle we can modify the structure.